source/opencascade/boundary_lib.cc

```diff
@@ -233,13 +233,13 @@ namespace OpenCASCADE
   ArclengthProjectionLineManifold<dim,spacedim>::pull_back(const Point<spacedim> &space_point) const
   {
     double t (0.0);
-#ifdef DEBUG
     ShapeAnalysis_Curve curve_analysis;
     gp_Pnt proj;
-    Assert(curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true)
-           < tolerance*length, ExcPointNotOnManifold(space_point));
+#ifdef DEBUG
+    Assert(curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true) <
+           tolerance*length, ExcPointNotOnManifold(space_point));
 #else
-    (void)space_point;
+    curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true);
```

Comment:
To make the code easier to read, I'd much rather have something like
```
  result = computation();
  Assert (result == ..., Exc...());
#ifndef DEBUG
  (void)result;
#endif
```
At the very least, it is clear that one has to change only one place if the `computation()` call needs to be adjusted later on. The style where you write it twice simply invites a divergence of code.